src/scheduler.py

```diff
@@ -371,7 +377,17 @@ def _verify_architecture_filter(self, job, node):
             return False
         return True
 
-    def _run(self, sub_id):
+    def _run(self, sub_ids):
+        threads = []
+        for sub_id in sub_ids:
+            thread = threading.Thread(target=self._run_scheduler, args=(sub_id,))
```

Comment:
both threads call _api_helper concurrently.
Are we sure kernelci.api_helper is thread-safe?